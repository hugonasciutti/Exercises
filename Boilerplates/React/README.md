# Features

Dry boiler plate with:

- Webpack dev server
- React
- Babel Env
