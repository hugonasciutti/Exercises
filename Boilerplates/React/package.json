{
  "name": "boilerplatereact",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "webpack-dev-server"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "final-form": "^4.5.2",
    "react": "^16.3.2",
    "react-dom": "^16.3.2"
  },
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.4",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "html-webpack-plugin": "^3.2.0",
    "webpack": "^4.6.0",
    "webpack-cli": "^2.0.15",
    "webpack-dev-server": "^3.1.3"
  }
}
