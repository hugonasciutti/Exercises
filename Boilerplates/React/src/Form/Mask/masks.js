const hourInSeconds = 3600
const minuteInSeconds = 60

function joinMask(mask) {
  return new RegExp(mask.map(regex => regex.source || regex).join(''))
}

function formatHoursToSeconds(value) {
  const result = (/^(\d{2}):(\d{2})$/g).exec(value)

  if (result) {
    let seconds = (parseFloat(result[1]) * hourInSeconds)
    seconds += (parseFloat(result[2]) * minuteInSeconds)

    return seconds
  }

  return value
}

function formatHoursToString(number) {
  const numbersStartsWithZero = 10

  let hours = Math.floor(number / hourInSeconds)
  let minutes = Math.floor((number - (hours * hourInSeconds)) / minuteInSeconds)

  if (hours < numbersStartsWithZero) {
    hours = `0${hours}`
  }

  if (minutes < numbersStartsWithZero) {
    minutes = `0${minutes}`
  }

  return `${hours}:${minutes}`
}

function formatSecondsToHours(number) {
  const zero = 0

  if (number === zero) {
    return '00:00'
  }

  if (number && !(/_/g).test(number)) {
    return formatHoursToString(number)
  }

  return number
}

function buildMask(mask, validationError) {
  return {
    mask,
    validationError: validationError || 'general.formatInvalid',
    validations: { matchRegexp: joinMask(mask) },
    formatter: value => formatHoursToSeconds(value),
    formatterOnReceive: value => formatSecondsToHours(value)
  }
}

export default {
  hour: buildMask([/[0-9]/, /[0-9]/, ':', /[0-5]/, /[0-9]/])
}
