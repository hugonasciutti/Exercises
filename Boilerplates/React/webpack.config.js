const HtmlWebpackPlugin = require('html-webpack-plugin')

module.exports = {
  mode: 'development',

  entry: './src/index.jsx',

  output: {
    path: `${__dirname}/dist`,
    filename: 'app.js'
  },

  devtool: 'source-map',

  module: {
    rules: [
      { test: /\.(js|jsx)?$/, loader: 'babel-loader', exclude: /node_modules/ }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      title: 'Boilerplate',
      template: './src/index.ejs'
    })
  ]
}
