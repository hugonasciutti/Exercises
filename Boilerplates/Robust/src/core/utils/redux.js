export const createReducer = (initialState, actions) =>
  (state = initialState, action) => {
    const reducer = actions[action.type]
    return reducer ? reducer(state, action) : state
  }

export default createReducer
