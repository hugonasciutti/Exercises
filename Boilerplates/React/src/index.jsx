import React, { Component } from 'react'
import { render } from 'react-dom'
import Form from './Form'

class App extends Component {
  render() {
    return (
      <div>
        <Input />
      </div>
    )
  }
}

render(<App />, document.getElementById('app'))