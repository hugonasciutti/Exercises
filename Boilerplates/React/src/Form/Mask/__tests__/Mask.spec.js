import React from 'react'
import { shallow } from 'enzyme'
import { buildEvent } from 'core/unit-test/utils'
import FormsyMask, { Mask } from '../'
import masks from '../masks'

describe('Mask', () => {
  it('should render component', () => {
    const mask = shallow(<Mask value="00:00" placeholder="xx" mask={masks.hour.mask} />)
    expect(mask).toMatchSnapshot()
  })

  it('should handler onChange', () => {
    const onChange = jest.fn()
    const mask = shallow(<Mask mask={masks.hour.mask} onChange={onChange} />)

    mask.simulate('change', buildEvent('00:00'))
    expect(onChange).toHaveBeenCalledWith('00:00')
  })

  it('should handler onChange using formatter', () => {
    const onChange = jest.fn()
    const formatter = hour => hour.replace(':', '')
    const mask = shallow(
      <Mask mask={masks.hour.mask} formatter={formatter} onChange={onChange} />
    )

    mask.simulate('change', buildEvent('00:00'))
    expect(onChange).toHaveBeenCalledWith('0000')
  })

  it('should remap props for formsy standard', () => {
    const mask = shallow(<FormsyMask mask="hour" />)
    expect(mask).toMatchSnapshot()
  })
})
