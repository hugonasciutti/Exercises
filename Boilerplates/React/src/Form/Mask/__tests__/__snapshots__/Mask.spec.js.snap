// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask should remap props for formsy standard 1`] = `
<Field
  formatter={[Function]}
  formatterOnReceive={[Function]}
  mask={
    Array [
      /\\[0-9\\]/,
      /\\[0-9\\]/,
      ":",
      /\\[0-5\\]/,
      /\\[0-9\\]/,
    ]
  }
  validationError="general.formatInvalid"
  validations={
    Object {
      "matchRegexp": /\\[0-9\\]\\[0-9\\]:\\[0-5\\]\\[0-9\\]/,
    }
  }
/>
`;

exports[`Mask should render component 1`] = `
<MaskedInput
  className="form-control "
  mask={
    Array [
      /\\[0-9\\]/,
      /\\[0-9\\]/,
      ":",
      /\\[0-5\\]/,
      /\\[0-9\\]/,
    ]
  }
  onChange={[Function]}
  placeholder="xx"
  value="00:00"
/>
`;
